### ISSUE TITLE
[SPIR-V] Shader containing discard statement without branching does not pass spirv validation

### ISSUE THREAD

**Issue by Gaspard--** [Microsoft/DirectXShaderCompiler#1250]
This bug turned up in a shader that was a result of a lot of preprocessing steps.
It seems a pixel shader that always discards does not compile.

### Minimal repro steps
```
void MainFp() {
    discard;
}
```
Compile the above with:
`dxc.exe -spirv -T ps_6_0 -E MainFp <filename>`

### Expected result
Succesful compilation.

### Actual result
The shader fails to compile with the following output:
```
fatal error: generated SPIR-V is invalid: Return must appear in a block
note: please file a bug report on https://github.com/Microsoft/DirectXShaderCompiler/issues with source code if possible
```

### Further technical details
The error  message tends to vary depending on what's around the discard statement. It always however resembles the format "XXX must appear in a block".
The shader compiles if the `discard` is in an if. The shader compiles under dxc and an older version of fxc.
I found this related pull request #546

**Comment by antiagainst** [comment]
Thanks for the bug report @Gaspard--! I've checked this and it seems spirv-opt is doing wrong transformations. I've filed https://github.com/KhronosGroup/SPIRV-Tools/issues/1527 for it.

**Comment by antiagainst** [comment]
This should be fixed by #1255.
